cfep-20.md

```diff
@@ -0,0 +1,97 @@
+
+<table>
+<tr><td> Title </td><td> Standard package split (dev / doc / dbg) </td>
+<tr><td> Status </td><td> Proposed </td></tr>
+<tr><td> Author(s) </td><td> Sylvain Corlay &lt;sylvain.corlay@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Oct 21, 2020</td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+This CFEP proposes a policy on how to split packages in multiple outputs for
+
+ - the main runtime requirements (shared objects, executables)
+ - the developer requirements (headers, static libraries, pkg-config and cmake artifacts)
+ - the requirements for debugging (debug symbols, JS map files)
+ - documentation.
+
+Notes:
+
+ - The Debian community follows a convention similar to this proposal.
+ - Should this CFEP be accepted, it would supersede [CFEP 18](https://github.com/conda-forge/cfep/blob/master/cfep-18.md) on the split of static libraries.
+ - This CFEP will be much easier to execute upon if we improve the conda tooling around the support for multiple outputs for recipes.
+
+## Motivation
+
+At the moment, there is not a common pattern to split packages in multiple output including C/C++ headers, JS source maps, or artifacts of build systems that are not required at runtime but are necessary at build time. Packages have been split in various ways, but adopting a convention will allow us to create tooling, idioms, and best practices on how to deal with such split packages.
+
```

Comment:
Again text specifically on which packages this applies to would be good. Do we want a `numpy-dev` for its headers? Are we going to rename `pybind11` to `pybind11-dev`?

Reply:
> Again text specifically on which packages this applies to would be good.

Yes, probably for all.

> numpy

numpy's headers are not in the standard include directory, but in package data. Also they are "needed" at runtime since there is a runtime function returning them, so I don't think there is a need for numpy-dev.

> Are we going to rename pybind11 to pybind11-dev

Yes, for the most part (but pybind11 is not quite header-only) so there should probably be both a pybind11 (with the python part and the package-data duplication of the headers), and pybind11-dev with the headers and the cmake things.